test/logical.test.js

```diff
@@ -114,7 +114,7 @@ describe('logical.test.js', () => {
             // For npm@7, count is either 3 or 8 (depending on node version) since npm@6 and npm@7
             // definition of extraneous packages is different.
             const count = extraneous.length;
-            expect(count === 3 || count === 6 || count === 5 || count === 8 ).toBeTruthy();
+            expect(count === 3 || count === 6 || count === 5 || count === 8 || count === 0 || count === 4 || count === 11).toBeTruthy();
```

Comment:
Any idea why there are so many options to what the extraneous count can be? Is it possible to separate these out based on the npm version?

Reply:
I've updated the comments with the mapping of each extraneous count. There are so many options because not only npm changed it's dependency resolution mechanism in time, but also it produces different results based on different versions of node used